src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final double side;
+
+    public Square(String color, double side) {
+        super(color);
+        this.side = side;
+    }
+
+    @Override
+    public double calculateArea() {
+        return side * side;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: square,"
+                + " area: " + calculateArea()
+                + " sq. units, side: " + side
+                + " units, color: " + getColor());
+    }
```

Comment:
The `draw()` method and the `calculateArea()` method should not be in the same class according to the checklist. Consider separating these responsibilities into different interfaces or classes.